### PR TITLE
fix RegExp to work with large files, removed unneeded lastIndex reset

### PR DESCRIPTION
**Pull request by Delagen**
as workaround on `Maximum call stack size exceeded` error with large files

### CODE REVIEW

**Comment by thlorenz** [comment]
Hi, thanks for the PR, but I'm having a hard time understanding what this actually does for the following reasons:

- lots of white space changes (please remove)
- no explanation in the description of how this works and why it couldn't reuse the code used for [fromSource](https://github.com/thlorenz/convert-source-map#fromsourcesource-largesource)
- seems like code duplicated instead of just reused (i.e. you could pull the code you need into a separate function)

Please fix as many of the above as you can to make it easier for me to review.
Thanks.

**Comment by Delagen** [comment]
@thlorenz Thanks for request
I made code review and found issue.
Slightly updated RegExp expression and error is gone. Also removed old code for largeFile support for `fromSource`. 
All tests passed.

**Comment by Delagen** [comment]
Tested on 8MB file from https://bugs.chromium.org/p/chromium/issues/detail?id=485052.
Both `fromSource` and `removeComments` fine.

**Comment by thlorenz** [comment]
Sorry no bandwidth for this this week, let's pick this up next week.
I'm gonna do a more detailed review and finally have a look at the other PRs as well.

Could you please explain in more detail why you had to remove 0.8 due to some inline-source-map thing?
Also please don't add anymore unrelated changes to this PR.
If more things come up (not related to the _remove_ API change) consider opening separate PRs.

Thanks for the work so far.

**Comment by Delagen** [comment]
Upgrade of ` inline-source-map` makes remove of 0.8 version of `nodejs` required because `source-map` 0.6.x which is dependency of it requires node>=0.10

**Comment by Delagen** [comment]
Three weeks past :(

File: index.js [hunk 1]
```diff
@@ -65,7 +55,7 @@ Converter.prototype.toBase64 = function () {
 
 Converter.prototype.toComment = function (options) {
   var base64 = this.toBase64();
-  var data = 'sourceMappingURL=data:application/json;base64,' + base64;
+  var data = 'sourceMappingURL=data:application/json;charset=utf-8;base64,' + base64;
```
Comment (thlorenz) [inline index.js]: I'm a little bit concerned about this change.
It's not directly related to the goal of your PR.

Could you please link me to resources that show this works in all browsers?

**Comment by thlorenz** [comment]
Sorry @Delagen. I'm very busy with other stuff.
Please address the issue regarding the `utf-8` change.

I also made you collaborator.

Once we clear that last question please merge to master following [this guide](https://gist.github.com/thlorenz/f7df0daf13c6f6c1fa84).
I will then version and publish to npm.

I believe due to the utf8 change it needs to be a major upgrade since it may break in odd cases.

Thanks.

File: .travis.yml [hunk 2]
```diff
@@ -1,5 +1,8 @@
 language: node_js
 node_js:
-  - 0.8
-  - 0.10
-  - 0.11
\ No newline at end of file
+  - "0.10"
+  - "0.11"
```
Comment (thlorenz) [inline .travis.yml]: Let's stick to even versions here, i.e. 0.12, 4, 6

File: index.js (hunk 1 above)
Comment (Delagen) [inline index.js]: This is from Generator.prototype.inlineMappingUrl of your project `inline-source-map` that adds charset even if it not specified

Comment (Delagen) [inline index.js]: to refrect test
```
t.equal(convert.fromComment(comment).toComment(), comment, 'comment -> comment')
```
where comment
```
comment = gen.inlineMappingUrl()
```

File: .travis.yml (hunk 2 above)
Comment (Delagen) [inline .travis.yml]: OK

File: index.js (hunk 1 above)
Comment (thlorenz) [inline index.js]: OK sounds good, thanks

**Comment by thlorenz** [comment]
LGTM, please merge to master and I'll publish a new version to npm.

**Comment by Delagen** [comment]
OK

**Comment by thlorenz** [comment]
Sorry should've told you to squash to one commit per feature before merging .. I'll do that tomorrow and force push to master .. no need to have all details of this addition in the history.

**Comment by Delagen** [comment]
I thought about squash, but in merge guide no info about it. So made as in guide

**Comment by thlorenz** [comment]
Yeah, I understand, may need to add that to the guide .. no worries I'll take care of it today and publish a new version.

**Comment by thlorenz** [comment]
OK, squashed, force pushed and published. Thanks for your help

**Comment by phated** [comment]
I'm pretty sure this broke stuff in `gulp-sourcemaps`.

My comment from the commit that broke stuff:
> @thlorenz This commit is really weird. It looks like a reversion but removed the lastIndex changes to the RegExps which breaks code that uses .exec on the RegExps exported by the module.
Ref https://github.com/gulp-sourcemaps/gulp-sourcemaps/blob/master/src/init/index.internals.js#L99

**Comment by Delagen** [comment]
I think that if you use Regexp from other module you must control lastIndex in your code.

**Comment by phated** [comment]
You still broke the API in a minor version

**Comment by Delagen** [comment]
RTFM api does not specified that RegExp resetted on every get

**Comment by phated** [comment]
That's the whole point of the getters. Definitely read all the f***ing docs and code. Definitely frustrated that you broke userland in this hacky patch.

**Comment by thlorenz** [comment]
@phated sorry I didn't see the lastIndex thing (what happens when you got too many modules to maintain).
However please calm your language so we can figure out a good solution here.

From what I saw there was no API change, just how things worked internally. Seems like gulp dependent on internal workings. How hard is it to fix this in gulp?

@Delagen how important is the change to not reset the regex? I'm thinking we could publish a patch to make it work as it did before?

**Comment by Delagen** [comment]
@thlorenz lastIndex stuff can be reverted, but it doesn't solve possible   issues in dependent projects.
So code
```
var regexp=convertSourceMap.commentRegExp;

function a() {
...
regexp.exec();
...
}
function b() {
...
regexp.exec();
...
}
```
has issues, so dev's mustn't depend on internals, and write predictable code

**Comment by thlorenz** [comment]
> If we revert code we doesn't solve static problem

How about we revert the code, but allow doing what you're trying to do via a flag or by passing an option somewhere?

**Comment by Delagen** [comment]
You can place lastIndex stuff in getters. Moreover you an author and you can do all as you want. I simply specified that this does not prevent error when developer doesn.t care about it in own code

**Comment by phated** [comment]
@thlorenz apologies for the language; I don't take kindly to being told to RTFM. 

I think there's a pretty easy solution. I'll send a PR with tests this afternoon.

**Comment by thlorenz** [comment]
Wow, missed the _RTFM_ thing originally (just had to look up what it means) .. so your reaction makes more sense now.
Anyways looking forward to the PR, we'll publish it as a patch which then should fix the affected modules.

**Pull request merged**